### ISSUE TITLE
Inconsistent naming conventions for types

### ISSUE THREAD

**Issue by benfrankel** [bevyengine/bevy#14274]
Checked the naming conventions for many "types of things" in Bevy.

Inconsistent:

- `SystemSet`: `System`, `Set`, `Systems`, and no-suffix
- `ScheduleLabel`: `Schedule` and no-suffix (odd one out: `ExtractSchedule`)
- `RenderLabel`: `Label` and no-suffix (odd one out: `CameraDriverLabel`)
- `SystemParam`: `Param` and no-suffix (odd one out: `UiLayoutSystemRemovedComponentParam`)
- `Event`: `Event` and no-suffix
- `Primitive2d`: `2d`, `2`, and no-suffix
- `Primitive3d`: `3d`, `3`, `3A`, and no-suffix

Consistent:

- `Plugin`: `Plugin`
- `PluginGroup`: `Plugins` (`PluginGroupBuilder` does not count)
- `Bundle`: `Bundle`
- `AppLabel`: `App`
- `System`: `System`
- `Material`: `Material`
- `Component`: no-suffix
- `Resource`: no-suffix
- `Command`: no-suffix
- `Asset`: no-suffix

N/A:

- `States`

Hard to search for because no distinguishing trait:

- Observable events
- Marker components

**Comment by benfrankel** [comment]
`Event`, `Primitive2d`, and `Primitive3d` are probably "fake inconsistent" and fine to stay as they are. So this can be narrowed down to `SystemSet` and three odd-one-out type names.

**Comment by alice-i-cecile** [comment]
Can you explain more about what you think should be done in all of the inconsistent cases?

**Comment by benfrankel** [comment]
Trivial cases:

- `ScheduleLabel`: Rename `ExtractSchedule` -> `RenderExtract` or `ExtractRender`
- `RenderLabel`: Rename `CameraDriverLabel` -> `CameraDriver`
- `SystemParam`: Rename `UiLayoutSystemRemovedComponentParam` -> `UiLayoutSystemRemovedComponent`
- `Event`: Leave as-is
- `Primitive2d`: Leave as-is
- `Primitive3d`: Leave as-is

Non-trivial case:

- `SystemSet`: Decide between `Systems`, `System`, `Set`, or no-suffix. Rename all system sets accordingly.

## `FooSystems`

Pros:
- Parallels `Plugins` for `PluginGroup`
- Unambiguous

Cons:
- Longest option (besides `SystemSet` as a suffix...)
- Used by 2/20 bevy system set types: `SimulationLightSystems` and `VisibilitySystems`
- Rarely (never?) used by 3rd-party crates

## `FooSystem`

Pros:
- 1 character shorter than `Systems`
- Used by 10/20 bevy system set types.
- Used by 3rd-party crates sometimes

Cons:
- Still long
- Ambiguous (is it a system or a system set?)

## `FooSet`

Pros:
- Very short
- Commonly used by 3rd-party crates

Cons:
- Used by 2/20 bevy system set types: `RenderSet` and `ExtractMeshesSet`
- Ambiguous (is it a `SystemSet` or a set of `Foo`?)

**Comment by benfrankel** [comment]
I'm leaning `Set` for system sets, but it's not obvious.

**Comment by janhohenheim** [comment]
I fully agree with the suggestions presented here. They seem reasonable and applicable. Especially the `SystemSet` consistency is a potential documentation hazard since users might not know what to search for.

**Comment by Atlas16A** [comment]
Could make `SystemSet` short via `SysSet` since everyone knows Sys is abbreviation for system

**Comment by benfrankel** [comment]
A Discord poll resulted in 26-1-27-3 votes for `FooSystems` vs `FooSystem` vs `FooSet` vs `Foo`.
